src/my_crypto.erl

```diff
@@ -19,17 +19,38 @@ is_public_key(Key) ->
 
 %% Actual function implementations
 
-hash_full(Bin) -> crypto:hash(sha256, Bin).
+hash_full(Bin) -> 
+    Full = crypto:hash(sha256, Bin),
 
-sign_full(Hash, PrivateKey) -> public_key:sign(Hash, none, PrivateKey).
+    %% Copied this online :cat_scream:
+    HexString = lists:flatten([io_lib:format("~2.16.0B",[X]) || <<X:8>> <= Full ]),
 
-verify_full(Hash, Signature, PubKey) -> public_key:verify(Hash, none, Signature, PubKey).
+    list_to_binary(HexString).
 
-read_file_key_full(private, FileName) ->
-    {ok, PemBin} = file:read_file(FileName),
+%% hash_full(Bin) -> 
+%%     crypto:hash(sha256, Bin).
+
+sign_full(Hash, PrivateKey) -> 
+    {ok, PemBin} = file:read_file(PrivateKey),
     [RSAEntry] = public_key:pem_decode(PemBin),
-    Key = public_key:pem_entry_decode(RSAEntry),
-    Key;
+    ActualPrivateKey = public_key:pem_entry_decode(RSAEntry),
+    public_key:sign(Hash, none, ActualPrivateKey).
+
+%% sign_full(Hash, PrivateKey) -> 
+%%     public_key:sign(Hash, none, PrivateKey).
+
+verify_full(Hash, Signature, PubKey) -> 
+    public_key:verify(Hash, none, Signature, PubKey).
+
+read_file_key_full(private, FileName) ->
+    {ok, _PemBin} = file:read_file(FileName),
+    FileName;
```

Comment:
😱  what is the point of this function.

Reply:
at this point none, but I wanna keep it this way in case we are changing it later. kind of black box/api thing

Reply:
also I was lazy

Reply:
the real reason

Reply:
oh, but it verifies that the file exists

Reply:
i guess i could also verify it's a valid private key

Reply:
but again. it's good enough and im lazy

Reply:
How does this function verify if the file exists or not?

Reply:
by reading it :cat_scream:

Reply:
oh wait I see, I already forgot how erlang works.